pkg/store/copr/region_cache.go

```diff
@@ -176,10 +176,20 @@ func (c *RegionCache) SplitKeyRangesByLocationsWithBuckets(bo *Backoffer, ranges
 		if limit != UnspecifiedLimit && len(res) >= limit {
 			break
 		}
-		loc, err := c.LocateKey(bo.TiKVBackoffer(), ranges.At(0).StartKey)
-		if err != nil {
-			return res, derr.ToTiDBErr(err)
+
+		loc := c.TryLocateKey(ranges.At(0).StartKey)
+		if loc == nil {
+			// load the regions in the key range into region cache, we don't use the returned locations for minimal change.
+			_, err := c.LoadRegionsInKeyRange(bo.TiKVBackoffer(), ranges.RefAt(0).StartKey, ranges.RefAt(0).EndKey)
```

Comment:
It may introduce a lot of `insertRegionToCache` with `invalidateOldRegion=true`, I'm afraid this might cause some problems.

Reply:
Good catch, unfornately the function `LoadRegionsInKeyRange` don't accept options now.

What do you think if adding the options to every exposed function in region cache.

```go
type LocateConfiguration struct {
	invalidateOldRegion bool
}

type LocateOption func(*LocateConfiguration)

func WithInvalidateOldRegion(invalidate bool) LocateOption {
	return func(c *LocateConfiguration) {
		c.invalidateOldRegion = invalidate
	}
}

func (c *RegionCache) LocateRegionByID(bo *retry.Backoffer, regionID uint64, opts ...LocateOption) 
func (c *RegionCache) LoadRegionsInKeyRange(bo *retry.Backoffer, startKey, endKey []byte, opts ...LocateOption) 
...
```

Reply:
LGTM